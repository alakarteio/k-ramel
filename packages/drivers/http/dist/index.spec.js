/* eslint-env jest */
import driver from './index.es'
import tests from '../misc/test-suite'

tests(driver)
