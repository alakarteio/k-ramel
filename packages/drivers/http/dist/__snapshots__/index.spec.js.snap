// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`drivers http should fetch data an trigger common events 1`] = `
Object {
  "dispatch": Array [
    Array [
      Object {
        "fetch": Array [
          "http://google.fr",
          Object {
            "headers": Object {},
          },
        ],
        "payload": undefined,
        "status": undefined,
        "type": "@@http/GOOGLE>GET>STARTED",
      },
    ],
    Array [
      Object {
        "type": "DISPATCHED",
      },
    ],
    Array [
      Object {
        "fetch": undefined,
        "payload": Object {
          "json": [Function],
        },
        "status": undefined,
        "type": "@@http/GOOGLE>GET>ENDED",
      },
    ],
  ],
  "fetch": Array [
    Array [
      "http://google.fr",
      Object {
        "headers": Object {},
      },
    ],
  ],
}
`;

exports[`drivers http should fetch data and calls json() 1`] = `
Object {
  "dispatch": Array [
    Array [
      Object {
        "fetch": Array [
          "http://google.fr",
          Object {
            "headers": Object {},
          },
        ],
        "payload": undefined,
        "status": undefined,
        "type": "@@http/GOOGLE>GET>STARTED",
      },
    ],
    Array [
      Object {
        "type": "DISPATCHED",
      },
    ],
    Array [
      Object {
        "fetch": undefined,
        "payload": Object {
          "result": "http://google.fr",
        },
        "status": undefined,
        "type": "@@http/GOOGLE>GET>ENDED",
      },
    ],
  ],
  "fetch": Array [
    Array [
      "http://google.fr",
      Object {
        "headers": Object {},
      },
    ],
  ],
}
`;

exports[`drivers http should fetch data and dispatch a FAILED 1`] = `
Object {
  "dispatch": Array [
    Array [
      Object {
        "fetch": Array [
          "http://google.fr",
          Object {
            "headers": Object {},
          },
        ],
        "payload": undefined,
        "status": undefined,
        "type": "@@http/GOOGLE>GET>STARTED",
      },
    ],
    Array [
      Object {
        "type": "DISPATCHED",
      },
    ],
    Array [
      Object {
        "fetch": undefined,
        "payload": Object {
          "status": 404,
          "url": "http://google.fr",
        },
        "status": 404,
        "type": "@@http/GOOGLE>GET>FAILED",
      },
    ],
  ],
  "fetch": Array [
    Array [
      "http://google.fr",
      Object {
        "headers": Object {},
      },
    ],
  ],
}
`;

exports[`drivers http should safely fail to parse json 1`] = `
Object {
  "dispatch": Array [
    Array [
      Object {
        "fetch": Array [
          "http://google.fr",
          Object {
            "headers": Object {},
          },
        ],
        "payload": undefined,
        "status": undefined,
        "type": "@@http/GOOGLE>GET>STARTED",
      },
    ],
    Array [
      Object {
        "type": "DISPATCHED",
      },
    ],
    Array [
      Object {
        "fetch": Array [
          "http://google.fr",
          Object {
            "headers": Object {},
          },
        ],
        "payload": [Error: oups-json-http://google.fr],
        "status": undefined,
        "type": "@@http/GOOGLE>GET>FAILED",
      },
    ],
  ],
  "fetch": Array [
    Array [
      "http://google.fr",
      Object {
        "headers": Object {},
      },
    ],
  ],
}
`;

exports[`drivers http should safely fail to request 1`] = `
Object {
  "dispatch": Array [
    Array [
      Object {
        "fetch": Array [
          "http://google.fr",
          Object {
            "headers": Object {},
          },
        ],
        "payload": undefined,
        "status": undefined,
        "type": "@@http/GOOGLE>GET>STARTED",
      },
    ],
    Array [
      Object {
        "fetch": Array [
          "http://google.fr",
          Object {
            "headers": Object {},
          },
        ],
        "payload": [Error: oups-http://google.fr],
        "status": undefined,
        "type": "@@http/GOOGLE>GET>FAILED",
      },
    ],
    Array [
      Object {
        "type": "DISPATCHED",
      },
    ],
  ],
  "fetch": Array [
    Array [
      "http://google.fr",
      Object {
        "headers": Object {},
      },
    ],
  ],
}
`;

exports[`drivers http should set customs options 1`] = `
Object {
  "fetchCounts": 2,
  "globalOptions": Array [
    "http://google.fr",
    Object {
      "credentials": "include",
      "headers": Object {
        "Authorization": "Bearer <my-token>",
        "other": "header",
      },
      "other": "option",
    },
  ],
  "overridedOptions": Array [
    "http://google_2.fr",
    Object {
      "credentials": "new credentials",
      "headers": Object {
        "Authorization": "new auth",
        "other": "header",
        "yep": "new one header",
      },
      "other": "option",
      "yep": "new one option",
    },
  ],
}
`;

exports[`drivers http should stringify data but not override content-type 1`] = `
Object {
  "dispatch": Array [
    Array [
      Object {
        "fetch": Array [
          "http://google.fr",
          Object {
            "body": "{\\"some\\":\\"data\\"}",
            "headers": Object {
              "Content-Type": "my-contentType",
            },
            "method": "POST",
          },
        ],
        "payload": undefined,
        "status": undefined,
        "type": "@@http/GOOGLE>POST>STARTED",
      },
    ],
    Array [
      Object {
        "type": "DISPATCHED",
      },
    ],
    Array [
      Object {
        "fetch": undefined,
        "payload": Object {
          "options": Object {
            "body": "{\\"some\\":\\"data\\"}",
            "headers": Object {
              "Content-Type": "my-contentType",
            },
            "method": "POST",
          },
          "url": "http://google.fr",
        },
        "status": undefined,
        "type": "@@http/GOOGLE>POST>ENDED",
      },
    ],
  ],
  "fetch": Array [
    Array [
      "http://google.fr",
      Object {
        "body": "{\\"some\\":\\"data\\"}",
        "headers": Object {
          "Content-Type": "my-contentType",
        },
        "method": "POST",
      },
    ],
  ],
}
`;

exports[`drivers http should stringify data with method helper 1`] = `
Object {
  "dispatch": Array [
    Array [
      Object {
        "fetch": Array [
          "http://google.fr",
          Object {
            "body": "{\\"some\\":\\"data\\"}",
            "headers": Object {
              "Content-Type": "application/json",
            },
            "method": "POST",
          },
        ],
        "payload": undefined,
        "status": undefined,
        "type": "@@http/GOOGLE>POST>STARTED",
      },
    ],
    Array [
      Object {
        "type": "DISPATCHED",
      },
    ],
    Array [
      Object {
        "fetch": undefined,
        "payload": Object {
          "options": Object {
            "body": "{\\"some\\":\\"data\\"}",
            "headers": Object {
              "Content-Type": "application/json",
            },
            "method": "POST",
          },
          "url": "http://google.fr",
        },
        "status": undefined,
        "type": "@@http/GOOGLE>POST>ENDED",
      },
    ],
  ],
  "fetch": Array [
    Array [
      "http://google.fr",
      Object {
        "body": "{\\"some\\":\\"data\\"}",
        "headers": Object {
          "Content-Type": "application/json",
        },
        "method": "POST",
      },
    ],
  ],
}
`;

exports[`drivers http should works with method helper (OPTIONS) 1`] = `
Object {
  "dispatch": Array [
    Array [
      Object {
        "fetch": Array [
          "http://google.fr",
          Object {
            "headers": Object {},
            "method": "OPTIONS",
          },
        ],
        "payload": undefined,
        "status": undefined,
        "type": "@@http/GOOGLE>OPTIONS>STARTED",
      },
    ],
    Array [
      Object {
        "type": "DISPATCHED",
      },
    ],
    Array [
      Object {
        "fetch": undefined,
        "payload": Object {
          "options": Object {
            "headers": Object {},
            "method": "OPTIONS",
          },
          "url": "http://google.fr",
        },
        "status": undefined,
        "type": "@@http/GOOGLE>OPTIONS>ENDED",
      },
    ],
  ],
  "fetch": Array [
    Array [
      "http://google.fr",
      Object {
        "headers": Object {},
        "method": "OPTIONS",
      },
    ],
  ],
}
`;
